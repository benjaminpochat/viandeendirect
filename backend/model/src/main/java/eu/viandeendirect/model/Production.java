package eu.viandeendirect.model;

import java.net.URI;
import java.util.Objects;

import com.fasterxml.jackson.annotation.*;
import eu.viandeendirect.model.BeefProduction;
import eu.viandeendirect.model.HonneyProduction;
import eu.viandeendirect.model.PackageLot;
import eu.viandeendirect.model.Producer;
import eu.viandeendirect.model.Sale;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import org.openapitools.jackson.nullable.JsonNullable;
import java.time.OffsetDateTime;
import jakarta.validation.Valid;
import jakarta.validation.constraints.*;
import io.swagger.v3.oas.annotations.media.Schema;


import java.util.*;
import jakarta.annotation.Generated;

/**
 *
 */

@Schema(name = "Production", description = "")
@JsonIgnoreProperties(
  value = "productionType", // ignore manually set productionType, it will be automatically generated by Jackson during serialization
  allowSetters = true // allows the productionType to be set during deserialization
)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "productionType", visible = true)
@JsonSubTypes({
  @JsonSubTypes.Type(value = BeefProduction.class, name = "BeefProduction"),
  @JsonSubTypes.Type(value = HonneyProduction.class, name = "HonneyProduction")
})

@Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
@jakarta.persistence.Entity @jakarta.persistence.Inheritance(strategy = jakarta.persistence.InheritanceType.TABLE_PER_CLASS)
public abstract class Production {

  @JsonProperty("id")
  @jakarta.persistence.Id
  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "production_id_generator")
  @SequenceGenerator(name="production_id_generator", sequenceName = "production_id_seq", allocationSize = 1)
  private BigDecimal id;

  /**
   * Gets or Sets productionType
   */
  public enum ProductionTypeEnum {
    BEEFPRODUCTION("BeefProduction"),

    HONNEYPRODUCTION("HonneyProduction");

    private String value;

    ProductionTypeEnum(String value) {
      this.value = value;
    }

    @JsonValue
    public String getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    @JsonCreator
    public static ProductionTypeEnum fromValue(String value) {
      for (ProductionTypeEnum b : ProductionTypeEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }
  }

  @JsonProperty("productionType")
  private ProductionTypeEnum productionType;

  @JsonProperty("sales")
  @jakarta.persistence.ManyToMany(mappedBy = "productions")
  @Valid
  private List<Sale> sales = null;

  @JsonProperty("producer")
  @ManyToOne
  @JsonBackReference
  private Producer producer;

  @JsonProperty("lots")
  @jakarta.persistence.OneToMany(mappedBy = "production")
  @Valid
  private List<PackageLot> lots = null;

  public Production id(BigDecimal id) {
    this.id = id;
    return this;
  }

  /**
   *
   * @return id
  */
  @Valid
  @Schema(name = "id", description = "", required = false)
  public BigDecimal getId() {
    return id;
  }

  public void setId(BigDecimal id) {
    this.id = id;
  }

  public Production productionType(ProductionTypeEnum productionType) {
    this.productionType = productionType;
    return this;
  }

  /**
   * Get productionType
   * @return productionType
  */

  @Schema(name = "productionType", required = false)
  public ProductionTypeEnum getProductionType() {
    return productionType;
  }

  public void setProductionType(ProductionTypeEnum productionType) {
    this.productionType = productionType;
  }

  public Production sales(List<Sale> sales) {
    this.sales = sales;
    return this;
  }

  public Production addSalesItem(Sale salesItem) {
    if (this.sales == null) {
      this.sales = new ArrayList<>();
    }
    this.sales.add(salesItem);
    return this;
  }

  /**
   *
   * @return sales
  */
  @Valid
  @Schema(name = "sales", description = "", required = false)
  public List<Sale> getSales() {
    return sales;
  }

  public void setSales(List<Sale> sales) {
    this.sales = sales;
  }

  public Production producer(Producer producer) {
    this.producer = producer;
    return this;
  }

  /**
   * Get producer
   * @return producer
  */
  @Valid
  @Schema(name = "producer", required = false)
  public Producer getProducer() {
    return producer;
  }

  public void setProducer(Producer producer) {
    this.producer = producer;
  }

  public Production lots(List<PackageLot> lots) {
    this.lots = lots;
    return this;
  }

  public Production addLotsItem(PackageLot lotsItem) {
    if (this.lots == null) {
      this.lots = new ArrayList<>();
    }
    this.lots.add(lotsItem);
    return this;
  }

  /**
   *
   * @return lots
  */
  @Valid
  @Schema(name = "lots", description = "", required = false)
  public List<PackageLot> getLots() {
    return lots;
  }

  public void setLots(List<PackageLot> lots) {
    this.lots = lots;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Production production = (Production) o;
    return Objects.equals(this.id, production.id) &&
        Objects.equals(this.productionType, production.productionType) &&
        Objects.equals(this.sales, production.sales) &&
        Objects.equals(this.producer, production.producer) &&
        Objects.equals(this.lots, production.lots);
  }

  @Override
  public int hashCode() {
    return Objects.hash(id, productionType, sales, producer, lots);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Production {\n");
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    productionType: ").append(toIndentedString(productionType)).append("\n");
    sb.append("    sales: ").append(toIndentedString(sales)).append("\n");
    sb.append("    producer: ").append(toIndentedString(producer)).append("\n");
    sb.append("    lots: ").append(toIndentedString(lots)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

