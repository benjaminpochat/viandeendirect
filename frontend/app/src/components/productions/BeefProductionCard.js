import { useEffect, useState } from 'react';
import { useKeycloak } from '@react-keycloak/web'
import { Button, Card, CardActions, CardContent, Typography } from "@mui/material"
import { AuthenticatedApiBuilder } from '../security/AuthenticatedApiBuilder'
import dayjs from 'dayjs'

function BeefProductionCard({ production }) {

    const [beefProduction, setBeefProduction] = useState(production)
    const { keycloak, initialized } = useKeycloak()
    const authenticatedApiBuilder = new AuthenticatedApiBuilder()

    useEffect(() => {
        let api = authenticatedApiBuilder.getAuthenticatedApi(keycloak);
        authenticatedApiBuilder.invokeAuthenticatedApi(() => {
            api.getBeefProduction(production.id, (error, data, response) => {
                if (error) {
                    console.error(error);
                } else {
                    console.log('api.getBeefProduction called successfully. Returned data: ' + data);
                    setBeefProduction(data)
                }
            })
        }, keycloak)
    }, [keycloak])

    return (
        <Card>
            <CardContent>
                <Typography color="text.secondary" gutterBottom>
                Abattage bovin
                </Typography>
                <Typography variant="h5" component="div">
                Abattage le {dayjs(beefProduction.slaughterDate).format('DD/MM/YYYY')}
                </Typography>
                <Typography component="div">
                Poids vif estim√© : {beefProduction.animalLiveWeight} kg
                </Typography>
            </CardContent>
            <CardActions>
                <Button size="small">Mettre en vente</Button>
                <Button size="small">Ajuster le poids vif</Button>
            </CardActions>
        </Card>
    )
}

export default BeefProductionCard