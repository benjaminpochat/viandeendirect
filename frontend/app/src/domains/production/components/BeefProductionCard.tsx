import React from 'react';
import { Button, Card, CardActionArea, CardActions, CardContent, Typography } from "@mui/material"
import { useEffect, useState } from 'react'
import { useKeycloak } from '@react-keycloak/web'
import { ApiBuilder } from '../../../api/ApiBuilder.ts'
import dayjs from 'dayjs'

export default function BeefProductionCard({
    production: production, 
    showActions: showActions, 
    viewBeefProductionCallback: viewBeefProductionCallback}) {

    const [beefProduction, setBeefProduction] = useState(production)
    const { keycloak, initialized } = useKeycloak()
    const apiBuilder = new ApiBuilder()

    useEffect(() => {
        apiBuilder.getAuthenticatedApi(keycloak).then(api => {
            apiBuilder.invokeAuthenticatedApi(() => {
                api.getBeefProduction(production.id, (error, data, response) => {
                    if (error) {
                        console.error(error);
                    } else {
                        console.log('api.getBeefProduction called successfully. Returned data: ' + data);
                        setBeefProduction(data)
                    }
                })
            }, keycloak)
        })
    }, [keycloak])

    return (
        <Card>
            <CardActionArea onClick={() => viewBeefProductionCallback(beefProduction)}>
                <CardContent>
                    <Typography color="text.secondary" gutterBottom>
                        Abattage bovin
                    </Typography>
                    <Typography variant="subtitle1" component="div">
                        Abattage le {dayjs(beefProduction.slaughterDate).format('DD/MM/YYYY')}
                    </Typography>
                    <Typography component="div">
                        Poids de carcasse chaude : {beefProduction.warmCarcassWeight} kg
                    </Typography>
                </CardContent>
            </CardActionArea>
            {getActions()}
        </Card>
    )

    function getActions() {
        if (showActions) {
            return <CardActions>
                <Button size="small">Mettre en vente</Button>
            </CardActions>
        }
    }
}
