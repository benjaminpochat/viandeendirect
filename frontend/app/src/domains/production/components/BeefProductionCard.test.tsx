import renderer from 'react-test-renderer';
import BeefProductionCard from './BeefProductionCard';
import React from 'react';
import { ReactKeycloakProvider } from '@react-keycloak/web';
import Production from 'viandeendirect_eu/dist/model/Production';

jest.mock("@react-keycloak/web", () => ({ useKeycloak: mockUseKeycloak }));


function mockUseKeycloak() {
  const token = "A random string that is non zero length";
  const userProfile: KeycloakProfile = {
    username: "test",
    email: "test@example.com",
    firstName: "Test",
    lastName: "User",
  };
  const realmAccess = { roles: ["user"] };

  const authClient: Keycloak = {
    authenticated: true,
    hasRealmRole(ignored: string) {
      return true;
    },
    hasResourceRole(ignored: string) {
      return true;
    },
    idToken: token,
    profile: userProfile,
    realm: "TestRealm",
    realmAccess,
    refreshToken: token,
    token,
  } as Keycloak;
  return { initialized: true, keycloak: authClient };
}

it('should render without error', () => {

  const production = {
    slaughterDate: '2024-02-01T12:00:00.000',
    warmCarcassWeight: 350
  }

  const component = renderer.create(
    <BeefProductionCard production={production} showActions={undefined} viewBeefProductionCallback={undefined} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
});