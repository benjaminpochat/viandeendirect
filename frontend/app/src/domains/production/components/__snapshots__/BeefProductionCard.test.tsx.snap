// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without error 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1ydyjjr-MuiTypography-root"
      >
        Abattage bovin
      </p>
      <div
        className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        Abattage le 
        01/02/2024
      </div>
      <div
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Poids de carcasse chaude : 
        350
         kg
      </div>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
